spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesClientInformerAutoConfiguration.java

```diff
@@ -0,0 +1,134 @@
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnCloudPlatform;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.cloud.CloudPlatform;
+import org.springframework.cloud.client.CommonsClientAutoConfiguration;
+import org.springframework.cloud.client.ConditionalOnDiscoveryEnabled;
+import org.springframework.cloud.client.discovery.simple.SimpleDiscoveryClientAutoConfiguration;
+import org.springframework.cloud.kubernetes.client.KubernetesClientAutoConfiguration;
+import org.springframework.cloud.kubernetes.commons.KubernetesNamespaceProvider;
+import org.springframework.cloud.kubernetes.commons.config.NamespaceResolutionFailedException;
+import org.springframework.cloud.kubernetes.commons.discovery.ConditionalOnKubernetesDiscoveryEnabled;
+import org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryProperties;
+import org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryPropertiesAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.log.LogAccessor;
+
+import static io.kubernetes.client.util.Namespaces.NAMESPACE_ALL;
+import static io.kubernetes.client.util.Namespaces.NAMESPACE_DEFAULT;
+import static org.springframework.cloud.kubernetes.client.KubernetesClientUtils.getApplicationNamespace;
+
+/**
+ * @author wind57
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnDiscoveryEnabled
+@ConditionalOnKubernetesDiscoveryEnabled
+@ConditionalOnBlockingOrReactiveEnabled
+@ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES)
+@AutoConfigureBefore({ SimpleDiscoveryClientAutoConfiguration.class, CommonsClientAutoConfiguration.class })
+@AutoConfigureAfter({ KubernetesClientAutoConfiguration.class, KubernetesDiscoveryPropertiesAutoConfiguration.class })
+public class KubernetesClientInformerAutoConfiguration {
+
+	private static final LogAccessor LOG = new LogAccessor(
+			LogFactory.getLog(KubernetesClientInformerAutoConfiguration.class));
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SharedInformerFactory sharedInformerFactory(ApiClient client) {
+		return new SharedInformerFactory(client);
+	}
+
+	@Bean
+	public String kubernetesClientNamespace(KubernetesDiscoveryProperties properties,
+			KubernetesNamespaceProvider provider) {
+		String namespace;
+		if (properties.allNamespaces()) {
+			namespace = NAMESPACE_ALL;
+			LOG.debug(() -> "serviceSharedInformer will use all-namespaces");
+		}
+		else {
+			try {
+				namespace = getApplicationNamespace(null, "kubernetes client discovery", provider);
+			}
+			catch (NamespaceResolutionFailedException ex) {
```

Comment:
this is the important distinction. We used to default to the `default` namespace `kubernetesNamespaceProvider.getNamespace() == null`. To preserve the sam functionality, I have this try/catch block. In a future release, this should fail with a `NamespaceResolutionFailedException`, like we do everywhere else